File: Java/game/src/main/java/com/yan/game/controller/GameController.java
package com.yan.game.controller;

import com.yan.game.dto.InfoDto;
import com.yan.game.service.IResultService;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/game")
public class GameController {

    private IResultService resultService;

    public GameController(IResultService resultService) {
        this.resultService = resultService;
    }

    @GetMapping("/userinfo/{userId}")
    public List<InfoDto> getUserInfo(@PathVariable Long userId) {
        return resultService.getUserInfo(userId);
    }

    @GetMapping("/levelinfo/{levelId}")
    public List<InfoDto> getLevelInfo(@PathVariable Long levelId) {
        return resultService.getLevelInfo(levelId);
    }

    @PutMapping("/setinfo")
    public InfoDto setInfo(@RequestBody InfoDto infoDto) {
        resultService.setInfo(infoDto);
        return infoDto;
    }
}

File: Kotlin/game/src/main/kotlin/com/yan/game/controller/GameController.kt
package com.yan.game.controller

import com.yan.game.dto.InfoDto
import com.yan.game.responses.Response
import com.yan.game.service.ResultService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/game")
class GameController @Autowired constructor(val resultService: ResultService) {

    @GetMapping("/userinfo/{userId}")
    fun getUserInfo(@PathVariable userId: Int): List<InfoDto> {
        return resultService.getUserInfo(userId)
    }

    @GetMapping("/levelInfo/{levelId}")
    fun getLevelInfo(@PathVariable levelId: Int): List<InfoDto> {
        return resultService.getLevelInfo(levelId)
    }

    @PutMapping("/setInfo")
    fun setInfo(@RequestBody infoDto: InfoDto): Response {
        return resultService.setInfo(infoDto)
    }
}